### DISCUSSION TITLE
First impressions / feedback

### DISCUSSION THREAD

**Discussion by ovizii** [circled-me/server#5] [Feedback]
So I ran the server successfully via docker, downloaded the Android app and used “create first user” to set up my account. I simply wanted to give feedback, if you like you can close this issue as soon as you have read it, but discussions seem disabled for this GitHub project, so I left the feedback here as “issue”.

I was a bit lost until I found the instruction:

> Now you can use the app and connect to your server at http://<YOUR_IP>:8080 and create your first admin user. Then you need to create a Storage Bucket from Settings and assign that to any User that is going to backup their photos there. If you used the example docker compose above, then the Path or your new Storage Bucket should be something like /mnt/data1/some-sub-dir

I am still confused about a few issues, let me list them here:

- when creating the first local storage bucket, I entered /local_test as path, and it ended up in the root of the container although in my docker-compose.yml I had mounted       `- ./data/assets:/mnt/data1` - I thought circled-server would know to save in there. I guess it doesn't matter what and where we mount, we just need to remember the path and create local storage with the right paths.

- while logged in as my first user, I can still access the “create first user” function, did you leave this here in case one is setting up a second server? I expected it to be gone as not to confuse users. Not sure what will happen if I set up a second “create first user” though...

- I have assigned myself the local storage nothing really changed, I gave myself a quota, the “0MB out of “smiley” still shows no quota.

- I was still stuck with “Albums” showing the text no albums... telling me how to create some... It wasn't until I forcefully closed the app that my library and Moments tabs showed up.

- my account still says: 0MB out of 490GB - at first I wasn't sure where this figure came from but that is the amount of available storage on my Docker library. This does not make much sense since I do not save anything in there, everything is supposed to be saved in mounted volumes, in this case in       `- ./data/assets:/mnt/data1` - I manually moved my local storage to be /mnt/data1/local_test and now quota looks right: 0MB out of 390GB except the 1000MB quota I assigned myself should be shown here, right?

Some screenshots showing the users quota and the account quota:

![image](https://github.com/circled-me/server/assets/837902/1f4b7611-804c-4e1d-9de3-3a9711955c14)

![image](https://github.com/circled-me/server/assets/837902/6d7d1c3a-4cfd-4e1c-aeb3-19d0595299b7)

I am also wondering about (1) and (2) as I can't find any docs on how it all works. Will try exploring on my own meanwhile.

- I tried activating the backup service and it immediately started uploading images from my phone without giving me any option of configuring what I want backed up. 

I often see this error in my logs:

```
circled-server  | 2023/10/15 10:08:16 /go/src/circled-server/handlers/album.go:64 record not found
circled-server  | [0.164ms] [rows:0] SELECT * FROM `favourite_assets` WHERE user_id = 1 ORDER BY `favourite_assets`.`user_id` LIMIT 1
circled-server  | [GIN] 2023/10/15 - 10:08:16 | 200 |     1.91527ms |     172.16.5.16 | GET      "/album/list"
circled-server  | [GIN] 2023/10/15 - 10:08:16 | 200 |  133.934834ms |     172.16.5.16 | GET      "/asset/fetch?thumb=1&size=500&id=10"
circled-server  | [GIN] 2023/10/15 - 10:08:16 | 200 |  140.654683ms |     172.16.5.16 | GET      "/asset/fetch?thumb=1&size=500&id=2"
circled-server  | [GIN] 2023/10/15 - 10:08:20 | 200 |    1.658978ms |     172.16.5.16 | GET      "/album/assets?album_id=2"
circled-server  | [GIN] 2023/10/15 - 10:08:21 | 200 |    2.125767ms |     172.16.5.16 | GET      "/asset/fetch?thumb=1&id=9"
circled-server  | [GIN] 2023/10/15 - 10:08:31 | 200 |    2.023726ms |     172.16.5.16 | GET      "/asset/fetch?thumb=1&id=8"
circled-server  | [GIN] 2023/10/15 - 10:08:34 | 200 |    2.803395ms |     172.16.5.16 | GET      "/asset/fetch?thumb=1&id=2"

```

**Comment by nikolaydimitrov** [comment]
Hi @ovizii and thanks for the thorough review!
I'll have a proper look in a couple of hours and give you details.

And thanks for pointing the lack of Discussions! I have just enabled it 👍🏼

**Comment by ovizii** [comment]
This way, you encourage discussions between users without polluting your issues with simple questions. You can also transform an issue into a discussion :-)

**Comment by nikolaydimitrov** [comment]
Sorry for the belated response! Moved this to a discussion :)

So, quick note about quotas - currently I made it so admins never have quota, even if you explicitly provide a number. But it's confusing and probably a good idea to be able to have quota indeed. I'm going to change that!
The space usage for admins was supposed to show the actual free space for the whole bucket, but I am just going to move that to the bucket's interface 👍🏼

1. About storage buckets location, the application inside the docker actually doesn't know what path you mounted your "external" disk. It can assume it is in `/mnt/data1`, but it could well not be there

2. Create first user is always available, because you can add many users to the same application (to different servers). Actually this is one of the good things about this - to be able to use multliple accounts at the same time.

3. The moments not appearing is really a bug, I am aware of it (now even more), so will try to fix it. As you said, this happens actually only for the very first time. Now you shouldn't have any problems with new 'Moments' or 'Albums'.

Thanks again for the feedback, much appreciated! 🙏
PS: Please always use `gubble:circled-server:latest`
